libraries/AR_Motors/AP_MotorsUGV.cpp

```diff
@@ -214,7 +226,58 @@ void AP_MotorsUGV::set_throttle(float throttle)
     }
 
     // check throttle is between -_throttle_max and  +_throttle_max
-    _throttle = constrain_float(throttle, -_throttle_max, _throttle_max);
+    float throttle_chk = constrain_float(throttle, -_throttle_max, _throttle_max);
+    _esc_backward_unlock = constrain_int16(_esc_backward_unlock,0,1);
```

Comment:
I don't think you need to check it every loop, its in the sanity check. Changing your if from `(_esc_backward_unlock != 0)` to `(_esc_backward_unlock > 0)` results in the same logic.

Reply:
done